README.md

```diff
@@ -345,3 +345,72 @@ w := kafka.NewWriter(kafka.WriterConfig{
 	Dialer:   dialer,
 })
 ```
+
+## SASL Support
+
+For a bare bones Conn type or in the Reader/Writer configs you can specify a dialer option for SASL authentication. If the `SASLMechanism` field is `nil`, it will not authenticate with SASL.
+
+There are implementations for [Plain](https://godoc.org/github.com/segmentio/kafka-go/sasl/plain#Mechanism)
+and [SCRAM](https://godoc.org/github.com/segmentio/kafka-go/sasl/scram#Mechanism) included.
+
+### Connection
```

Comment:
Do we want any examples of `Basic`?

Reply:
do you mean examples of plain (since i only included scram?) or something else

Reply:
Ah yes, sorry.  `Plain` is what I meant. :)

Reply:
added a section "SASL Authentication Types"